src/Microsoft.AspNetCore.Mvc.RazorPages/DependencyInjection/MvcRazorPagesMvcCoreBuilderExtensions.cs

```diff
@@ -58,6 +58,28 @@ public static IMvcCoreBuilder AddRazorPages(this IMvcCoreBuilder builder)
             return builder;
         }
 
+        /// <summary>
+        /// Configures Razor Pages to use the specified <paramref name="rootDirectory"/>.
+        /// </summary>
+        /// <param name="builder">The <see cref="IMvcCoreBuilder"/>.</param>
+        /// <param name="rootDirectory">The application relative path to use as the root directory.</param>
+        /// <returns></returns>
+        public static IMvcCoreBuilder WithRazorPagesRoot(this IMvcCoreBuilder builder, string rootDirectory)
+        {
+            if (builder == null)
+            {
+                throw new ArgumentNullException(nameof(builder));
+            }
+
+            if (string.IsNullOrEmpty(rootDirectory))
+            {
+                throw new ArgumentException(Resources.ArgumentCannotBeNullOrEmpty, nameof(rootDirectory));
+            }
```

Comment:
I was relying on the option setter to do this, but presumably this would be easier to debug.